mitodl/odl-video-service
# Video-specific permissions

#### What are the relevant tickets?
Closes #341 
Closes #342 

#### What's this PR do?
Adds video-specific permissions (including 'public') that override collection permissions.

#### How should this be manually tested?
- Set `ENABLE_VIDEO_PERMISSIONS=True` in your `.env` file.
  - Set collection view and admin permissions to certain levels (only me or moira lists) and verify that they still work as designed. (Superusers, collection owners and people on admin lists or view lists should be able to view the video).
  - Set video permissions to public.  Try to view in another browser as an anonymous user.  You should be able to view it.
  - Set video permissions to 'Only me'.  
    - You should be able to view it.  
    - Log in on another browser as someone who is on the collection's view lists (but not admin lists, not superuser).  You should not be able to view it.  
    - Log in as someone who is on the collection's admin lists.  You should be able to view it.
  - Set video permissions to 'Moira lists' and choose a list that is not in the collections view or admin lists.  
    - In another browser, log in as someone who is on that list, you should be able to view it.  
    - Log in as someone who is on one of the collection's admin lists, you should be able to view it.
    - Log in as someone who is on one of the collection's view lists, you should not be able to view it.
  - Switch the video permissions back to 'Same as collection'.  Verify that the collection's view permissions are being applied to the video.

- Set `ENABLE_VIDEO_PERMISSIONS=False` in your `.env` file.
  - The video-specific permissions options should not appear in the video edit form, and only collection-level permissions should be enforced.

Screenshot of edit video form (@roberthouse54 does this look ok to you):
<img width="878" alt="screen shot 2017-11-07 at 10 38 56 am" src="https://user-images.githubusercontent.com/187676/32502112-e4fc6460-c3a7-11e7-9694-596ba6b80bc9.png">
